cooked-validators/src/Cooked/Tx/Constraints/Type.hs

```diff
@@ -788,15 +786,17 @@ txSkelInputValue skel@TxSkel {txSkelMints = mints} =
   positivePart (txSkelMintsValue mints)
     <> foldOf (consumedOutputsF % sOutValueL) skel
 
+newtype Fee = Fee {feeLovelace :: Integer} deriving (Eq, Ord, Show, Num)
+
 -- | The value in all transaction inputs, plus the negative parts of the minted
 -- value. This is the right hand side of the "balancing equation":
 --
 -- > mints + inputs = fees + burns + outputs
-txSkelOutputValue :: TxSkel -> Pl.Value
-txSkelOutputValue skel@TxSkel {txSkelMints = mints} =
+txSkelOutputValue :: TxSkel -> Fee -> Pl.Value
```

Comment:
I have mixed feelings about this.

Indeed, having the fee as an explicit argument is a bit confusing and surprising. On the other hand, if one forgets a fee here, they'll get a compilation error. If they misread (or ignore) a warning in the comment, the compiler won't do anything (unfortunately!). Neither of the solutions feel completely satisfactory to me, but I'd rather err on the side of caution, where the compiler prevents me from making mistakes. Hence, I've decided to add an explicit `Fee` argument here.

In a sense, it's a way to get a reminder from the compiler — "hey, don't forget about fees in whatever calculation you're doing!", which is very easy to silence by putting a mere `0` in place of the argument (but this is something that's explicit).

In a sense, it's a question of the semantics of the function, and the meaning of "output value". To me, it's the fee as well, because, well, it's a part of the transaction, and one has to remember about this.